gen/target.cpp

```diff
@@ -84,14 +87,28 @@ llvm::Type *getRealType(const llvm::Triple &triple) {
 
   case Triple::ppc64:
   case Triple::ppc64le:
+    if (realPrecision == RealPrecision::DoubleDouble) {
+      return LLType::getPPC_FP128Ty(ctx);
+    }
     if (triple.isMusl()) { // Musl uses double
       return LLType::getDoubleTy(ctx);
     }
-    // dual-ABI complications: PPC only has IEEE 128-bit quad precision on
-    // Linux, IBM double-double is available on both AIX and Linux.
-    return triple.isOSLinux() && realTypeEncoding == RealTypeEncoding::IEEEQuad
-               ? LLType::getFP128Ty(ctx)
-               : LLType::getPPC_FP128Ty(ctx);
+#if defined(__linux__) && defined(__powerpc64__)
+    // for a PowerPC64 Linux build:
+    // default to the C++ host compiler's `long double` ABI
+    switch (std::numeric_limits<long double>::digits) {
```

Comment:
I would use:

```suggestion
    switch (__LDBL_MANT_DIG__) {
```

This one is defined by the compiler directly and I have tested to contain correct information.

Reply:
Fine by me. - We should probably only forward the host precision when targeting a *Linux* ppc64 target, not e.g. default to IEEE quad when cross-compiling to a ppc64 FreeBSD target.

Reply:
when changed to the compiler macro, could this be #if sequence such that failure happens during compilation rather than at runtime?